Sake/DecomposeMixer.cs

```diff
@@ -0,0 +1,83 @@
+﻿namespace Sake;
+
+/// <summary>
+/// https://github.com/lontivero/DecompositionsPlayground/blob/master/Notebook.ipynb
+/// </summary>
+public static class Decomposer
+{
+	public static long[] StdDenoms;
+
+	public static IEnumerable<(long Sum, int Count, ulong Decomposition)> Decompose(long target, long tolerance, int maxCount)
+	{
+		var denoms = StdDenoms.SkipWhile(x => x > target).ToArray();
+
+		return denoms.SelectMany((_, i) => InternalCombinations(target, tolerance: tolerance / 10, maxCount, denoms)).Take(10000).ToList();
+	}
+
+	private static IEnumerable<(long Sum, int Count, ulong Decomposition)> InternalCombinations(long target, long tolerance, int maxLength, long[] denoms)
+	{
+		IEnumerable<(long Sum, int Count, ulong Decomposition)> Combinations(
+			int currentDenominationIdx,
+			ulong accumulator,
+			long sum,
+			int k)
+		{
+			accumulator = (accumulator << 8) | ((ulong)currentDenominationIdx & 0xff);
+			var currentDenomination = denoms[currentDenominationIdx];
+			sum += currentDenomination;
+			var remaining = target - sum;
+			if (k == 0 || remaining < tolerance)
+				return new[] { (sum, maxLength - k, accumulator) };
+
+			currentDenominationIdx = Search(remaining, denoms, currentDenominationIdx);
+
+			return Enumerable.Range(0, denoms.Length - currentDenominationIdx)
+				.TakeWhile(i => k * denoms[currentDenominationIdx + i] >= remaining - tolerance)
+				.SelectMany((_, i) =>
+					Combinations(currentDenominationIdx + i, accumulator, sum, k - 1)
+					.TakeUntil(x => x.Sum == target));
+		}
+
+		return denoms.SelectMany((_, i) => Combinations(i, 0ul, 0, maxLength - 1)).Take(50).ToList();
```

Comment:
Here is where we limit how deep we want to search. There is no correct or wrong value and it can be only discovered empirically. The number 50 was used because the maxLength decomposition was 8 and the there were billions of solutions but now, it much harder to find solutions that improve the naive one so, this number can be 500.
```suggestion
		return denoms.SelectMany((_, i) => Combinations(i, 0ul, 0, maxLength - 1)).Take(500).ToList();
```

Reply:
I'll experiment with this.